h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -165,19 +252,34 @@ public void cv_computeAndSetOptimalParameters(ModelBuilder[] cvModelBuilders) {
       int lidx = 0; // index into submodel
       int bestId = 0;   // submodel indedx with best Deviance from xval
       int cnt = 0;
+      int alphaIndex = 0;
       for (; lidx < lmin_max; ++lidx) { // search through submodel with same lambda and alpha values
         double testDev = 0;
         double testDevSq = 0;
+
+        // Lambdas are sorted in decreasing order for same alpha values
+        if (lambdas[lidx] > lambdas[Math.max(lidx-1, 0)]) {
+          alphaIndex++;
+          Log.info(">>> Alpha["+lidx+"] _state.alpha: "+ _state.alpha());
+        }
+
         for (int i = 0; i < cvModelBuilders.length; ++i) {  // run cv for each lambda value
           GLM g = (GLM) cvModelBuilders[i];
-          if (g._model._output._submodels[lidx] != null) {
-            double lambda = g._model._output._submodels[lidx].lambda_value;
-            g._insideCVCheck = true;
-            g._driver.computeSubmodel(lidx, lambda, Double.NaN, Double.NaN);
-            g._insideCVCheck = false;
-            testDev += g._model._output._submodels[lidx].devianceValid;
-            testDevSq += g._model._output._submodels[lidx].devianceValid * g._model._output._submodels[lidx].devianceValid;
+          if (g._model._output._submodels[lidx] == null) {
```

Comment:
Using the same logic as before (https://github.com/h2oai/h2o-3/blob/229b0066382ad94b2021f1a790a992655ef0e095/h2o-algos/src/main/java/hex/glm/GLM.java#L124-L125)